content/criando-novos-comandos-no-django-admin.md

```diff
@@ -0,0 +1,395 @@
+        ''' faz a validação de ... '''
+        False if kwargs['imdbRating'] == "N/A" else True
+        False if kwargs['imdbVotes'] == "N/A" else True
+
+    def save(self, **kwargs):
+        try:
+            ''' verifica as validações '''
+            if self.validate_values(**kwargs):
+                ''' Então remove a virgula de imdbVotes '''
+                kwargs['imdbVotes'] = kwargs['imdbVotes'].replace(',', '')
+                ''' SALVA os dados '''
+                Movie.objects.create(**kwargs)
+        except ValidationError as e:
+            self.print_red(e.messages)
+            self.print_red('O objeto não foi salvo.\n')
+
+    def handle(self, movies, year, **options):
+        ''' se "movies" não for nulo, transforma em inteiro '''
+        if movies is not None:
+            movies = int(movies)
+        ''' busca os filmes n vezes, a partir da variavel "movies" '''
+        for i in range(movies):
+            m = self.get_movie(year)
+            ''' Transforma "year" em inteiro '''
+            if "–" in m['Year']:
+                m['Year'] = year
+            data = {
+                "title": m['Title'],
+                "year": m['Year'],
+                "released": m['Released'],
+                "director": m['Director'],
+                "actors": m['Actors'],
+                "poster": m['Poster'],
+                "imdbRating": m['imdbRating'],
+                "imdbVotes": m['imdbVotes'],
+                "imdbID": m['imdbID'],
+            }
+            self.save(**data)
+            print('\n', i + 1, data['year'], data['title'])
+
+        print('\nForam salvos %d filmes' % movies)
+```
+
+**Uso**
+
+```bash
+Usage: ./manage.py initdata [options] 
+
+Faz o crawler numa api de filmes e retorna os dados.
+     Uso: ./manage.py initdata
+     ou: ./manage.py initdata -m 20
+     ou: ./manage.py initdata -m 20 -y 2015
+```
+
+
+### search.py
+
+**Objetivo**: Localizar o filme pelo título ou ano de lançamento.
+
+```python
+from django.core.management.base import BaseCommand, CommandError
+from optparse import make_option
+from django.db.models import Q
+from core.models import Movie
+
+
+class Command(BaseCommand):
+    help = "Localiza um filme pelo título ou ano de lançamento.\n \
+    Uso: ./manage.py search -t 'Ted 2'\n \
+    ou: ./manage.py search -y 2015\n \
+    ou: ./manage.py search -t 'a' -y 2015"
+    option_list = BaseCommand.option_list + (
+        make_option('--title', '-t',
+                    dest='title',
+                    default=None,
+                    help='Localiza um filme pelo título.'),
+        make_option('--year', '-y',
+                    dest='year',
+                    default=None,
+                    help='Localiza um filme pelo ano de lançamento.'),
+    )
+
+    def handle(self, title=None, year=None, **options):
+        if title is not None and year is not None:
+            q = Movie.objects.filter(Q(title__istartswith=title) &
```

Comment:
Você não precisa do `Q` se vai somente construir um filtro com `AND`. 
`Movie.objects.filter(title__istartswith=title, year=year)`